persist/fs/retriever.go

```diff
@@ -523,6 +537,10 @@ func (req *retrieveRequest) onCallerOrRetrieverDone() {
 	}
 	req.id.Finalize()
 	req.id = nil
+	if req.tags != nil {
+		req.tags.Close()
+		req.tags = nil
```

Comment:
should this be `req.tags = ident.EmptyTagIterator`

Reply:
Sure thing (we initialize this with ident.EmptyTagIterator when we reset for reuse but we can do that here too).